<H1> PROJET 3 </H1>



In [0]:
import pandas as pd
import numpy as np

<B> <span> 1ère question : CALCUL DE LA POPULATION MONDIALE  </span>  </B>

In [2]:
data = pd.read_csv("FAOSTAT_data_POPULATION.csv")
data.columns = ['Code Domaine','Domaine','Code Pays','Pays','Code Élément','Élément','Code Produit','Produit','Code Année','Année','Unité','Valeur','Symbole','Description du Symbole']

FileNotFoundError: ignored

In [0]:
# Je ne garde que les colonnes qui m'intéressent(Je ne veux pas le Code Domaine,Domaine,Code Élément,Élément,Code Produit,Produit,Code Année,Année,Symbole,Description du Symbole )

df_population = data[['Code Pays','Pays','Unité','Valeur']]

In [0]:
# D'abord une 1ère analyse de mon DataFrame
df_population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 4 columns):
Code Pays    175 non-null int64
Pays         175 non-null object
Unité        175 non-null object
Valeur       175 non-null int64
dtypes: int64(2), object(2)
memory usage: 5.5+ KB


Il y a donc 175 lignes, je choisis comme clé primaire le code pays, et je vais supprimer les doublons pour vérifier qu'il y a toujours le même nombre de ligne après

In [0]:
#je choisis comme clé primaires le code pays et le vérifie en supprimant les doublons

df_population.drop_duplicates('Code Pays')

Code Pays                                    Pays           Unité  \
0            2                             Afghanistan  1000 personnes   
1          202                          Afrique du Sud  1000 personnes   
2            3                                 Albanie  1000 personnes   
3            4                                 Algérie  1000 personnes   
4           79                               Allemagne  1000 personnes   
5            7                                  Angola  1000 personnes   
6            8                      Antigua-et-Barbuda  1000 personnes   
7          194                         Arabie saoudite  1000 personnes   
8            9                               Argentine  1000 personnes   
9            1                                 Arménie  1000 personnes   
10          10                               Australie  1000 personnes   
11          11                                Autriche  1000 personnes   
12          52                             Azerbaïdjan  1000 personnes   
13          12                                 Bahamas  1000 personnes   
14          16                              Bangladesh  1000 personnes   
15          14                                 Barbade  1000 personnes   
16          57                                 Bélarus  1000 personnes   
17         255                                Belgique  1000 personnes   
18          23                                  Belize  1000 personnes   
19          53                                   Bénin  1000 personnes   
20          17                                Bermudes  1000 personnes   
21          19         Bolivie (État plurinational de)  1000 personnes   
22          80                      Bosnie-Herzégovine  1000 personnes   
23          20                                Botswana  1000 personnes   
24          21                                  Brésil  1000 personnes   
25          26                       Brunéi Darussalam  1000 personnes   
26          27                                Bulgarie  1000 personnes   
27         233                            Burkina Faso  1000 personnes   
28          35                              Cabo Verde  1000 personnes   
29         115                                Cambodge  1000 personnes   
..         ...                                     ...             ...   
145        195                                 Sénégal  1000 personnes   
146        272                                  Serbie  1000 personnes   
147        197                            Sierra Leone  1000 personnes   
148        199                               Slovaquie  1000 personnes   
149        198                                Slovénie  1000 personnes   
150        276                                  Soudan  1000 personnes   
151         38                               Sri Lanka  1000 personnes   
152        210                                   Suède  1000 personnes   
153        211                                  Suisse  1000 personnes   
154        207                                Suriname  1000 personnes   
155        209                               Swaziland  1000 personnes   
156        208                             Tadjikistan  1000 personnes   
157         39                                   Tchad  1000 personnes   
158        167                           Tchéquie (la)  1000 personnes   
159        216                               Thaïlande  1000 personnes   
160        176                             Timor-Leste  1000 personnes   
161        217                                    Togo  1000 personnes   
162        220                       Trinité-et-Tobago  1000 personnes   
163        222                                 Tunisie  1000 personnes   
164        213                            Turkménistan  1000 personnes   
165        223                                 Turquie  1000 personnes   
166        230                                 Ukraine  1000 personnes   
167      

In [0]:
#Je vérifie que le nombre de ligne est le même avant et après cette manipulation
df_population.size == df_population.drop_duplicates('Code Pays').size

True

Il y a toujours 175 lignes donc je garde Code Pays comme clé primaire

In [0]:
print((df_population["Valeur"].sum())/1000000, "milliard de personnes")
# je divise par 1 millions, car l'unité et de 1000 personnes, et je souhaite un résultats en milliards

8.413993 milliard de personnes


En 2013, la population mondiale était d'à peu près 7.083 milliard, nous avons donc 1.33 milliard de trop


In [0]:
# Je recherche le pays qui présenterait une redondance

df_population.loc[df_population['Pays'].str.contains('Chine')]

Code Pays                       Pays           Unité   Valeur
33          96   Chine - RAS de Hong-Kong  1000 personnes     7204
34         128       Chine - RAS de Macao  1000 personnes      566
35          41        Chine, continentale  1000 personnes  1385567
36         214  Chine, Taiwan Province de  1000 personnes    23330
174        351                      Chine  1000 personnes  1416667

In [0]:
# Je vérifie si "Chine"(174) est bien une agrégation des 4 régions
((df_population.iloc[[33,34,35,36], :])["Valeur"].sum()) ==((df_population.iloc[[174], :])["Valeur"].sum())

True

In [0]:
# Je décide donc de supprimer la ligne "Chine" pour garder les régions différenciées
df_population = df_population.drop(df_population.index[174])

In [0]:
# Je recalcule la population mondiale
print((df_population["Valeur"].sum())/1000000, "milliard de personnes")
# je divise par 1 millions, car l'unité et de 1000 personnes, et je souhaite un résultats en milliards

6.997326 milliard de personnes


Nous sommes donc à peu près à 7 milliards d'humains, le chiffre est donc tout à fait probable.

<B> <span> 2ème question : IDENTIFIER LES REDONDANCES  </span>  </B>

disponibilité intérieure = Production + importations - exportations + évolution des stocks

Mais je cherche aussi les emplois, je vais vérifier que l'on peut aussi dire : disponibilité intérieure = aliments pour animaux + semences + pertes + traitements + autres utilisations + nourriture

In [0]:
# J'importe mon nouveau DataFrame complet du bilan alimentaire sans l'agrégation "Chine"

df_bilan = pd.read_csv("FAOSTAT_BILAN_ALIMENTAIRE_COMPLET.csv")
df_bilan.columns = ['Code Domaine','Domaine','Code Pays','Pays','Code Élément','Élément','Code Produit','Produit','Code Année','Année','Unité','Valeur','Symbole','Description du Symbole']


In [0]:
# Je ne garde que les colonnes qui m'intéressent(Je ne veux pas le Code Domaine,Domaine )

df_bilan = df_bilan[['Code Pays','Pays','Code Élément','Élément','Code Produit','Produit','Code Année','Année','Unité','Valeur','Symbole','Description du Symbole']]

In [0]:
# Je vérifie la formule "disponibilité intérieure = Production + importations - exportations + évolution des stocks"
# avec la France et le blé

df_bilan.loc[(df_bilan['Pays'].str.contains('France')) & (df_bilan['Produit'].str.contains('Blé'))]

Code Pays    Pays  Code Élément  \
48648         68  France          5511   
48649         68  France          5611   
48650         68  France          5072   
48651         68  France          5911   
48652         68  France          5301   
48653         68  France          5521   
48654         68  France          5527   
48655         68  France          5123   
48656         68  France          5131   
48657         68  France          5154   
48658         68  France          5142   
48659         68  France           645   
48660         68  France           664   
48661         68  France           674   
48662         68  France           684   

                                                 Élément  Code Produit  \
48648                                         Production          2511   
48649                            Importations - Quantité          2511   
48650                                 Variation de stock          2511   
48651                            Exportations - Quantité          2511   
48652                           Disponibilité intérieure          2511   
48653                              Aliments pour animaux          2511   
48654                                           Semences          2511   
48655                                             Pertes          2511   
48656                                         Traitement          2511   
48657                                Autres Utilisations          2511   
48658                                         Nourriture          2511   
48659  Disponibilité alimentaire en quantité (kg/pers...          2511   
48660     Disponibilité alimentaire (Kcal/personne/jour)          2511   
48661  Disponibilité de protéines en quantité (g/pers...          2511   
48662  Disponibilité de matière grasse en quantité (g...          2511   

      Produit  Code Année  Année               Unité    Valeur Symbole  \
48648     Blé        2013   2013  Milliers de tonnes  38614.00       S   
48649     Blé        2013   2013  Milliers de tonnes   2055.00       S   
48650     Blé        2013   2013  Milliers de tonnes   1131.00       S   
48651     Blé        2013   2013  Milliers de tonnes  21502.00       S   
48652     Blé        2013   2013  Milliers de tonnes  20298.00       S   
48653     Blé        2013   2013  Milliers de tonnes   7822.00       S   
48654     Blé        2013   2013  Milliers de tonnes    748.00       S   
48655     Blé        2013   2013  Milliers de tonnes    358.00       S   
48656     Blé        2013   2013  Milliers de tonnes   1575.00       S   
48657     Blé        2013   2013  Milliers de tonnes   2824.00       S   
48658     Blé        2013   2013  Milliers de tonnes   6971.00       S   
48659     Blé        2013   2013                  kg    108.43      Fc   
48660     Blé        2013   2013  Kcal/personne/jour    811.00      Fc   
48661     Blé        2013   2013     g/personne/jour     25.99      Fc   
48662     Blé        2013   2013     g/personne/jour      4.20      Fc   

      Description du Symbole  
48648  Données standardisées  
48649  Données standardisées  
48650  Données standardisées  
48651  Données standardisées  
48652  Données standardisées  
48653  Données standardisées  
48654  Données standardisées  
48655  Données standardisées  
48656  Données standardisées  
48657  Données standardisées  
48658  Données standardisées  
48659        Donnée calculée  
48660        Donnée calculée  
48661        Donnée calculée  
48662        Donnée calculée

In [0]:
#Disponibilité intérieure = Production + importations + évolution des stocks - exportations 

(df_bilan.iloc[[48652], :]["Valeur"].sum()) == ((df_bilan.iloc[[48648,48649,48650], :])["Valeur"].sum())-((df_bilan.iloc[[48651], :])["Valeur"])

48651    True
Name: Valeur, dtype: bool

La formule est donc bien confirmée pour "disponibilité intérieure = Production + importations - exportations + évolution des stocks"

Maintenant essayons "disponibilité intérieure = aliments pour animaux + semences + pertes + traitements + autres utilisations + nourriture"

In [0]:
(df_bilan.iloc[[48652], :]["Valeur"].sum()) == ((df_bilan.iloc[[48653,48654,48655,48656,48657,48658], :])["Valeur"].sum())

True

<B> Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal </B>

In [0]:
# je veux créer un DataFrame ne contenant que la disponibilité alimentaire (Kcal/personne/jour) pour chaque produit

df_food_supply = df_bilan[ df_bilan['Code Élément'] == 664 ]

# Et je crée un DataFrame df_disponibilité contenant la disponibilité alimentaire + la population pour chaque pays (que je nettoie un peu)

df_disponibilité = pd.merge(df_food_supply, df_population, on='Code Pays', how='left')
df_disponibilité.drop(['Code Année', 'Pays_y', 'Symbole', 'Description du Symbole'], axis = 1, inplace = True) 


In [0]:
# Je crée une nouvelle colonne nommée "Dispo Kcal/Jour" qui est la multiplication de la disponibilité alimentaire 
# et de la population

df_disponibilité['Dispo Kcal/Jour'] = (df_disponibilité['Valeur_x'] * df_disponibilité['Valeur_y'] * 1000)
df_disponibilité

Code Pays       Pays_x  Code Élément  \
0              2  Afghanistan           664   
1              2  Afghanistan           664   
2              2  Afghanistan           664   
3              2  Afghanistan           664   
4              2  Afghanistan           664   
5              2  Afghanistan           664   
6              2  Afghanistan           664   
7              2  Afghanistan           664   
8              2  Afghanistan           664   
9              2  Afghanistan           664   
10             2  Afghanistan           664   
11             2  Afghanistan           664   
12             2  Afghanistan           664   
13             2  Afghanistan           664   
14             2  Afghanistan           664   
15             2  Afghanistan           664   
16             2  Afghanistan           664   
17             2  Afghanistan           664   
18             2  Afghanistan           664   
19             2  Afghanistan           664   
20             2  Afghanistan           664   
21             2  Afghanistan           664   
22             2  Afghanistan           664   
23             2  Afghanistan           664   
24             2  Afghanistan           664   
25             2  Afghanistan           664   
26             2  Afghanistan           664   
27             2  Afghanistan           664   
28             2  Afghanistan           664   
29             2  Afghanistan           664   
...          ...          ...           ...   
14211        181     Zimbabwe           664   
14212        181     Zimbabwe           664   
14213        181     Zimbabwe           664   
14214        181     Zimbabwe           664   
14215        181     Zimbabwe           664   
14216        181     Zimbabwe           664   
14217        181     Zimbabwe           664   
14218        181     Zimbabwe           664   
14219        181     Zimbabwe           664   
14220        181     Zimbabwe           664   
14221        181     Zimbabwe           664   
14222        181     Zimbabwe           664   
14223        181     Zimbabwe           664   
14224        181     Zimbabwe           664   
14225        181     Zimbabwe           664   
14226        181     Zimbabwe           664   
14227        181     Zimbabwe           664   
14228        181     Zimbabwe           664   
14229        181     Zimbabwe           664   
14230        181     Zimbabwe           664   
14231        181     Zimbabwe           664   
14232        181     Zimbabwe           664   
14233        181     Zimbabwe           664   
14234        181     Zimbabwe           664   
14235        181     Zimbabwe           664   
14236        181     Zimbabwe           664   
14237        181     Zimbabwe           664   
14238        181     Zimbabwe           664   
14239        181     Zimbabwe           664   
14240        181     Zimbabwe           664   

                                              Élément  Code Produit  \
0      Disponibilité alimentaire (Kcal/personne/jour)          2511   
1      Disponibilité alimentaire (Kcal/personne/jour)          2805   
2      Disponibilité alimentaire (Kcal/personne/jour)          2513   
3      Disponibilité alimentaire (Kcal/personne/jour)          2514   
4      Disponibilité alimentaire (Kcal/personne/jour)          2517   
5      Disponibilité alimentaire (Kcal/personne/jour)          2520   
6      Disponibilité alimentaire (Kcal/personne/jour)          2531   
7      Disponibilité alimentaire (Kcal/personne/jour)          2542   
8      Disponibilité alimentaire (Kcal/personne/jour)          2543   
9      Disponibilité alimentaire (Kcal/personne/jour)          2745   
10     Disponibilité alimentaire (Kcal/personne/jour)          2549   
11     Disponibilité alimentaire (Kcal/personne/jour)          2551   
12     Disponibilité alimentaire (Kcal/personne/jour)          2560   
13     Disponibilité alimentaire (Kcal/personne/jour)          2561   
14     Disponibilité alimen

<B> Calculez pour chaque produit le ratio "énergie/poids", que vous donnerez en kcal/kg </B>

In [0]:
# Il me faut donc le df_food_supply créé plus haut que je vais joindre à un DataFrame ne contenant que la 
# dispo alimentaire en quantité et créer une nouvelle colonne avec le ratio energie/poids

df_dispo_poids = df_bilan[ df_bilan['Code Élément'] == 645 ]
df_ratio_energie_poids = pd.merge(df_food_supply, df_dispo_poids, on=['Code Pays', 'Code Produit'], how='left')
df_ratio_energie_poids.drop(['Code Élément_x','Code Élément_y','Code Année_x', 'Pays_y','Code Élément_x','Symbole_y','Produit_y','Code Année_y','Année_y', 'Description du Symbole_y','Description du Symbole_x','Symbole_x'], axis = 1, inplace = True)
df_ratio_energie_poids['Ratio Energie/Poids'] = (df_ratio_energie_poids['Valeur_x'] / df_ratio_energie_poids['Valeur_y'])

# Je retire les lignes qui ont seulement la colonne ratio energie/poids = à NaN
df_ratio_energie_poids = df_ratio_energie_poids.dropna(subset = ['Ratio Energie/Poids'])

# et je retire les lignes où celui ci est égal à 0
df_ratio_energie_poids = df_ratio_energie_poids[df_ratio_energie_poids['Ratio Energie/Poids'] != 0]

df_ratio_energie_poids


Code Pays       Pays_x                                       Élément_x  \
0              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
1              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
2              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
3              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
4              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
6              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
7              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
8              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
9              2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
10             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
11             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
13             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
15             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
17             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
18             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
19             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
20             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
21             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
23             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
25             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
26             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
27             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
28             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
29             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
32             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
33             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
36             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
38             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
42             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
43             2  Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
...          ...          ...                                             ...   
14193        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14194        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14195        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14196        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14197        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14198        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14201        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14202        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14206        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14208        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14209        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14210        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14211        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14213        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14214        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14215        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14216        181     Zimbabwe  Disponibilité alimentaire (Kcal/personne/jour)   
14217        181     Zimbabwe  Disponi

<B>Donnez le ratio énergie/poids pour 5 produits de votre choix</B>

In [0]:
df_ratio_energie_poids.groupby('Produit_x')['Ratio Energie/Poids'].mean()

Produit_x
Abats Comestible             3.102668
Agrumes, Autres              0.704007
Aliments pour enfants       10.484800
Ananas                       1.212571
Animaux Aquatiques Autre     1.693765
Arachides Decortiquees      15.631442
Avoine                       6.228953
Bananes                      1.690458
Bananes plantains            2.232527
Beurre, Ghee                20.467195
Bière                        1.242396
Blé                          7.811005
Boissons Alcooliques         8.184981
Boissons Fermentés           1.341807
Café                         1.313890
Cephalopodes                 2.440512
Citrons & Limes              0.533778
Coco (Incl Coprah)           3.888541
Crustacés                    1.463991
Crème                        5.605604
Céréales, Autres             7.950039
Dattes                       5.922628
Edulcorants Autres           6.917432
Feve de Cacao               13.542626
Fruits, Autres               1.195673
Girofles                    10.151541
Gr

Par exemple :

Avoine                       6.228953
Bananes                      1.690458
Bananes plantains            2.232527
Beurre, Ghee                20.467195
Bière                        1.242396

<B> Calculez, pour les produits végétaux uniquement, la disponibilité intérieure mondiale exprimée en kcal </B>

In [0]:
#Pour ça il me faut un nouveau DF concernant les groupes de produits pris sur le site de la FAO (Sans l'agregation Chine)

df_groupe_produits = pd.read_csv("FAOSTAT_Groupe de produits.csv")
df_groupe_produits.drop(['Code Domaine','Domaine','Code Année','Année','Symbole','Description du Symbole'], axis = 1, inplace = True)

#Je crée un DF ne contenant que la disponibilité alimentaire et les végétaux :

df_végétaux = df_groupe_produits[(df_groupe_produits['Code Élément'] == 664 ) & ( df_groupe_produits['Code Produit'] == 2903) ]

# Et je crée un DataFrame df_disponibilité_végétaux contenant la disponibilité alimentaire des végétaux + la population pour chaque pays

df_disponibilité_végétaux = pd.merge(df_végétaux, df_population, on='Code Pays', how='left')

#Je crée une nouvelle colonne Dispo totale/jour

df_disponibilité_végétaux['Dispo totale/jour'] = (df_disponibilité_végétaux['Valeur_x'] * df_disponibilité_végétaux['Valeur_y'] * 1000)

# J'agrége toutes ces données pour avoir le total mondial en Kcal
print(df_disponibilité_végétaux["Dispo totale/jour"].sum(), "Kcal")




16582235067000.0 Kcal


<B> Question : en termes de disponibilité intérieure de produits végétaux, combien d'humains pourraient être nourris si toute la production végétale était utilisée pour de la nourriture ? </B>

In [0]:
#Recommandation FAO de 2414 Kcal/jour (je divise le résultats pour avoir en milliards)

print(((df_disponibilité_végétaux["Dispo totale/jour"].sum()/2414)/100000000),"milliards de personnes")

68.69194311101906 milliards de personnes
